R/filter_raw_reads.R

```diff
@@ -22,44 +22,70 @@ filter_raw_reads = function(
   id_cols=c('cell_set','treatment','dose','dose_unit','day','bio_rep','tech_rep'),
   reverse_index2 = FALSE) {
   
-  #insert profile_id into sample_meta
-  #sample_meta$profile_id = do.call(paste,c(sample_meta[id_cols], sep=':'))
-  
   # New: convert CB_meta from log10 to log2
   CB_meta= CB_meta %>% dplyr::mutate(log2_dose= log_dose/log10(2)) %>% dplyr::select(-log_dose)
   
-  if (reverse_index2){
+  if (reverse_index2) {
     sample_meta$IndexBarcode2 <- chartr("ATGC", "TACG", stringi::stri_reverse(sample_meta$IndexBarcode2))
     print("Reverse-complementing index 2 barcode.")
   }
   
   index_filtered = raw_counts %>%
-    dplyr::filter(index_1 %in% sample_meta$IndexBarcode1,
-                  index_2 %in% sample_meta$IndexBarcode2)
+    dplyr::filter(index_1 %in% sample_meta$IndexBarcode1, index_2 %in% sample_meta$IndexBarcode2)
   index_purity = sum(index_filtered$n) / sum(raw_counts$n)
-
+  
   cell_line_filtered = index_filtered %>%
     merge(sample_meta, by.x=c("index_1", "index_2"), by.y=c("IndexBarcode1", "IndexBarcode2")) %>%
     merge(cell_line_meta, by.x="forward_read_cl_barcode", by.y="Sequence", all.x=T) %>%
     merge(cell_set_meta, by="cell_set", all.x=T) %>%
     dplyr::filter(mapply(grepl, LUA, members) |
                     (mapply(grepl, LUA, cell_set) & is.na(members)) |
                     (forward_read_cl_barcode %in% CB_meta$Sequence))
-  cell_line_filtered$profile_id = do.call(paste,c(cell_line_filtered[id_cols], sep=':')) 
-  cell_line_purity = sum(cell_line_filtered$n) / sum(index_filtered$n)
-
+  cell_line_purity = sum(cell_line_filtered$n)/ sum(index_filtered$n)
+  
+  print("Generating QC table ...")
   qc_table = data.frame(cell_line_purity=cell_line_purity, index_purity = index_purity)
   
-  annotated_counts = cell_line_filtered %>%
+  # make template of expected reads
+  plate_map= sample_meta %>% dplyr::distinct(pick(c('IndexBarcode1', 'IndexBarcode2', 'pcr_plate', 'pcr_well')))
```

Comment:
Created a plate map so that all reads with valid indices will be mapped to a plate well. We can also add additional information such as everything from the sample meta.

Reply:
name barcode to plate, plate_map is used elsewhere